manifests/resource/upstream/member.pp

```diff
@@ -37,26 +37,11 @@
 define nginx::resource::upstream::member (
   $upstream,
   $server,
-  $ensure                 = 'present',
-  $port                   = 80,
+  Enum['present', 'absent'] $ensure = 'present',
```

Comment:
Since this parameter isn't used anymore, maybe default it to `undef` instead?

Then
```puppet
if $ensure { warning('DEPRECATION: some useful message here') }
```